server/src/main/java/com/genymobile/scrcpy/SurfaceEncoder.java

```diff
@@ -48,11 +49,34 @@ public SurfaceEncoder(SurfaceCapture capture, Streamer streamer, int videoBitRat
         this.downsizeOnError = downsizeOnError;
     }
 
+    /**
+     * Fixing the issue that the server crashes when requesting the maximum integer value(2147483647) for 'videoBitRate'
+     */
+    private void adjustMediaFormatValuesToSupported(MediaFormat format, MediaCodec mediaCodec) {
+        // Adjust VideoBitRate
+        try {
+            String mimeType = format.getString(MediaFormat.KEY_MIME);
+            if (mimeType != null) {
+                MediaCodecInfo mediaCodecInfo = mediaCodec.getCodecInfo();
+                Range<Integer> bitrateRange = mediaCodecInfo.getCapabilitiesForType(mimeType).getVideoCapabilities().getBitrateRange();
```

Comment:
In theory, it would be better to check all video capabilities like you are doing there.

But in the past, I noticed that in practice (on some devices), the reported values were not always accurate (especially for sizes), where "valid" values don't work or "invalid" values works. That's why I don't use them to filter parameters.

Reply:
As you mentioned, if there are cases where the attribute values provided by the equipment are invalid, a comprehensive test of the entire equipment (within a specific range) will be required.